plugins/item_tasks/server/rest.py

```diff
@@ -168,6 +168,30 @@ def _transformOutputs(self, outputs, token, job):
 
         return transformed
 
+    def _validateOutputParentType(self, outputId, parentType, outputSpec):
+        """
+        Checks if the output parent type is compatible with the output type.
+        """
+
+        # Find the corresponding output specification for the given outputID
+        output = None
+        for out in outputSpec:
+            if outputId == out['id']:
+                output = out
+                break
+
+        # If a corresponding output is found, check if its parent type is valid
+        if output:
```

Comment:
The for/else in Python is frequently misunderstood.  I'd avoid using it if possible.

https://aroberge.blogspot.com/2015/04/loopy-else-in-python.html